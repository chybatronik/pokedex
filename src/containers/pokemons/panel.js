import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Navbar, NavbarBrand, Nav, NavItem, NavLink } from 'reactstrap'
import { showWainting, showOnlyLoaded, showAll, SHOW_WAITING, SHOW_ONLY_LOADED } from '../../store/actions/filter'

class Panel extends Component {
  showWainting () {
    this.props.dispatch(showWainting())
  }

  showOnlyLoaded () {
    this.props.dispatch(showOnlyLoaded())
  }

  showAll () {
    this.props.dispatch(showAll())
  }

  render () {
    const { count, offset, filterActive } = this.props.pokemons.config
    const waiting = this.props.pokemons.waiting
    const isShowWaiting = filterActive === SHOW_WAITING
    const isShowOnlyLoaded = filterActive === SHOW_ONLY_LOADED
    const isShowAll = filterActive === null
    return (
      <Navbar fixed='top' sticky='top' color='faded' expand='md'>
        <NavbarBrand href='/'>pokemons : { count }</NavbarBrand>
        <Nav className='ml-auto' navbar>
          <NavItem>
            <NavLink href='#' disabled={isShowAll}  onClick={this.showAll.bind(this)}>all: {offset}</NavLink>
          </NavItem>
          <NavItem>
            <NavLink href='#' disabled={isShowOnlyLoaded} onClick={this.showOnlyLoaded.bind(this)}>loaded: {offset - waiting}</NavLink>
          </NavItem>
          <NavItem>
            <NavLink href='#' disabled={isShowWaiting} onClick={this.showWainting.bind(this)}>waiting : {waiting}</NavLink>
          </NavItem>
        </Nav>
      </Navbar>
    )
  }
}

const mapStateToProps = state => {
  const { pokemons } = state
  return {
    pokemons: {
      config: pokemons.config,
      waiting: Object.keys(state.fullPokemons).filter((item) => {return state.fullPokemons[item].isFetch}).length
    }
  }
}
export default connect(mapStateToProps)(Panel)
