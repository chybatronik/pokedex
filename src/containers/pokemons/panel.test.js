import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import { shallow } from 'enzyme'
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import { Panel } from './panel'

Enzyme.configure({ adapter: new Adapter() })

it('renders panel', () => {
  const output = shallow(
    <Panel
      config={{count: 11, offset: 77}}
      waiting={12}
    />
  )
  expect(shallowToJson(output)).toMatchSnapshot()
})
