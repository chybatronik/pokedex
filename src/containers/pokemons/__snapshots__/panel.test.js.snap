// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders panel 1`] = `
<Navbar
  color="faded"
  expand="md"
  fixed="top"
  sticky="top"
  tag="nav"
>
  <NavbarBrand
    href="/"
    tag="a"
  >
    pokemons : 
    11
  </NavbarBrand>
  <Nav
    className="ml-auto"
    navbar={true}
    tag="ul"
    vertical={false}
  >
    <NavItem
      tag="li"
    >
      <NavLink
        disabled={false}
        href="#"
        onClick={[Function]}
        tag="a"
      >
        all: 
        77
      </NavLink>
    </NavItem>
    <NavItem
      tag="li"
    >
      <NavLink
        disabled={false}
        href="#"
        onClick={[Function]}
        tag="a"
      >
        loaded: 
        65
      </NavLink>
    </NavItem>
    <NavItem
      tag="li"
    >
      <NavLink
        disabled={false}
        href="#"
        onClick={[Function]}
        tag="a"
      >
        waiting : 
        12
      </NavLink>
    </NavItem>
  </Nav>
</Navbar>
`;
